### PR TITLE
Nodejs/week3

### PR DESCRIPTION
**Pull request by wael85**

### CODE REVIEW

File: nodejs/week3/calculator/index.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const express = require('express');
+const app = express();
+app.get('/calculator/:method',(req,res)=>{
+    const firstParm =parseInt(req.query.firstParm);
```
Comment (benna100) [inline nodejs/week3/calculator/index.js:4]: Remember that there can be infinite query parameters!

File: nodejs/week3/calculator/index.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    const secoundParm =parseInt(req.query.secoundParm);
+    const method = req.params.method;
+    let result;
+    let methods = ['add','subtrac','multiply','divid'];
+    if (methods.indexOf(method)<0){
```
Comment (benna100) [inline nodejs/week3/calculator/index.js:9]: nice!

File: nodejs/week3/meal_share/src/backend/api/meals.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
+const express = require("express");
+const app = express();
+const router = express.Router();
+const knex = require("../database");
+const { sum } = require("../database");
+
+router.get("/", async (request, response) => {
+  
+  try {
+    let result = await knex(`meal`).select(`*`);
```
Comment (benna100) [inline nodejs/week3/meal_share/src/backend/api/meals.js:10]: Avoid variablenames like result. They add no information. describe what a variable contains, not what it is. This contains an array of meals. so `meals`

File: nodejs/week3/meal_share/src/backend/api/meals.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    const maxPrice = request.query.maxPrice;
```
Comment (benna100) [inline nodejs/week3/meal_share/src/backend/api/meals.js:11]: Look into using destructuring for this

File: nodejs/week3/meal_share/src/backend/api/meals.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    const availableReservations = request.query.availableReservations;
+    const title = request.query.title;
+    const createdAfter = request.query.createdAfter;
+    const limit = request.query.limit;
+    if(maxPrice){
+       result = await knex(`meal`).where(`price`,`<`,`${maxPrice}`).select(`*`);
+    }if(availableReservations){
+      result = await knex.select( "meal.id" , "meal.title").from("meal")
```
Comment (benna100) [inline nodejs/week3/meal_share/src/backend/api/meals.js:19]: nice

File: nodejs/week3/meal_share/src/backend/api/meals.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    }
+    response.json(result);
+  } catch (error) {
+    throw error;
+  }
+});
+router.post("/",async (req,res)=>{
+  try{
+    const insert = await knex('meal').insert(req.body);
+    res.send(insert);
```
Comment (benna100) [inline nodejs/week3/meal_share/src/backend/api/meals.js:41]: So what are you sending here?

File: nodejs/week3/meal_share/src/backend/api/reservation.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+      const result = await knex("reservation").where({id : id}).select("*");
+      response.json(result);
+    } catch (error) {
+      throw error;
+    }
+  });
+  router.put("/:id", async (request, response) => {
+    try {
+      const id = request.params.id;
+      const result = await knex("reservation").where({id : id}).update(request.body);
```
Comment (benna100) [inline nodejs/week3/meal_share/src/backend/api/reservation.js:33]: nice!

File: nodejs/week3/meal_share/src/backend/api/review.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+      const result = await knex("review").where({id : id}).select("*");
+      response.json(result);
+    } catch (error) {
+      throw error;
+    }
+  });
+  router.put("/:id", async (request, response) => {
+    try {
+      const id = request.params.id;
+      const result = await knex("review").where({id : id}).update(request.body);
```
Comment (benna100) [inline nodejs/week3/meal_share/src/backend/api/review.js:33]: Avoid the result variablename.

File: nodejs/week3/meal_share/src/backend/api/meals.js (hunk 6 above)
Comment (wael85) [reply nodejs/week3/meal_share/src/backend/api/meals.js:41]: I am getting back the id of the new meal I
 insert

**Pull request merged**